app/models/projects/exports/pdf_export/report.rb

```diff
@@ -0,0 +1,255 @@
+      if custom_field_select?(select)
+        process_custom_field_select(project, select, entries)
+      elsif project_phase_select?(select)
+        process_project_phase_select(project, select, entries)
+      elsif can_view_attribute?(project, select.attribute)
+        process_attribute_select(project, select, entries)
+      else
+        entries
+      end
+    end
+
+    def process_custom_field_select(project, select, entries)
+      return entries unless custom_field_active_in_project?(project, select.custom_field)
+
+      if select.custom_field.formattable?
+        write_table_entries(entries) unless entries.empty?
+        write_formattable_custom_field(project, select.custom_field)
+        []
+      else
+        entry = table_entry(project, "cf_#{select.custom_field.id}", select.caption)
+        entries.push(entry) if entry
+        entries
+      end
+    end
+
+    def process_project_phase_select(project, select, entries)
+      entry = user_can_view_project_phases?(project) ? table_entry_project_phase(project, select) : nil
+      entries.push(entry) if entry
+      entries
+    end
+
+    def process_attribute_select(project, select, entries)
+      if attribute_formattable?(select.attribute)
+        write_table_entries(entries) unless entries.empty?
+        write_formattable_attribute(project, select.attribute, select.caption)
+        []
+      else
+        entry = table_entry(project, select.attribute, select.caption)
+        entries.push(entry) if entry
+        entries
+      end
+    end
+
+    def project_phase_select?(select)
+      select.is_a?(::Queries::Projects::Selects::ProjectPhase)
+    end
+
+    def table_entry_project_phase(project, select)
+      phase = project.phases.active.find_by(definition: select.project_phase_definition)
+      return nil if phase.nil?
+
+      [
+        { content: select.caption }.merge(styles.project_attributes_table_label_cell),
+        format_phase_value(phase)
+      ]
+    end
+
+    def format_phase_value(phase)
+      start = if phase.start_date.present?
+                format_date(phase.start_date)
+              else
+                I18n.t("js.label_no_start_date")
+              end
+
+      finish = if phase.finish_date.present?
+                 format_date(phase.finish_date)
+               else
+                 I18n.t("js.label_no_due_date")
+               end
+
+      "#{start} - #{finish}"
+    end
+
+    def table_entry(project, value_name, caption)
+      value = format_attribute(project, value_name, :pdf)
+      return nil if hide_empty_attributes? && value.blank?
+
+      value = make_link_href_cell(url_helpers.project_url(project), value) if value_name == :id
+      [
+        { content: caption }.merge(styles.project_attributes_table_label_cell),
+        value || ""
+      ]
+    end
+
+    def can_view_attribute?(_project, attribute)
+      return false if attribute.nil? || %i[name favored].include?(attribute)
+
+      true
+    end
```

Comment:
I think the guard clause harms more than it helps in such a small method. Consider this:

```suggestion
    def can_view_attribute?(_project, attribute)
      attribute && !%i[name favored].include?(attribute)
    end
```

Reply:
Agreed, I remove the pointless guard.